src/pcapi/core/users/models.py

```diff
@@ -113,36 +97,65 @@ class NotificationSubscriptions:
     marketing_email: bool = True
 
 
+# calculate date of latest birthday
+def _get_latest_birthday(birth_date: date) -> date:
+    """
+    Calculates the latest birthday of a given person.
+    :param birth_date: The person's birthday.
+    :return: The latest birthday's date.
+    """
+
+    today = date.today()
+
+    # handle february 29th
+    if birth_date.month == 2 and birth_date.day == 29:
+        try:
+            this_year_birthday = date(today.year, birth_date.month, birth_date.day)
+        except ValueError:
+            this_year_birthday = date(today.year, 2, 28)
+        try:
+            previous_year_birthday = date(today.year - 1, birth_date.month, birth_date.day)
+        except ValueError:
+            previous_year_birthday = date(today.year - 1, 2, 28)
+    else:
+        this_year_birthday = date(today.year, birth_date.month, birth_date.day)
+        previous_year_birthday = date(today.year - 1, birth_date.month, birth_date.day)
```

Comment:
Puisque le 29/02 est le seul cas particulier, on peut probablement simplifier. Au passage, on peut utiliser `replace()` pour plus de lisibilité :

```suggestion
    try:
        this_year_birthday = birth_date.replace(year=today.year)
    except ValueError:  # handle February 29
        this_year_birthday = birth_date.replace(year=today.year, day=28)
    try:
        previous_year_birthday = birth_date.repace(year=today.year -1)
    except ValueError:  # handle February 29
        previous_year_birthday = birth_date.replace(year=today.year - 1, day=28)
```